### PR TITLE
support consent api

### PR DESCRIPTION
**Pull request by chaopeng**
Fixes https://github.com/GoogleCloudPlatform/healthcare-data-protection-suite/issues/776

### CODE REVIEW

File: templates/tfengine/components/resources/healthcare_datasets/main.tf [hunk 1]
```diff
@@ -23,6 +23,37 @@ module "{{resourceName . "name"}}" {
...
+    {{range .consent_stores -}}
+    {
+      name = "{{.name}}"
+      {{hclField . "iam_members" -}}
+
+      {{if has . "enable_consent_create_on_update" -}}
+      enable_consent_create_on_update = {
+        {{hcl .enable_consent_create_on_update}}
+      }
+      {{end -}}
```
Comment (xingao267) [inline templates/tfengine/components/resources/healthcare_datasets/main.tf:37]: You can do {{hclField .enable_consent_create_on_update}} for this.

File: templates/tfengine/components/resources/healthcare_datasets/main.tf [hunk 2]
```diff
@@ -23,6 +23,37 @@ module "{{resourceName . "name"}}" {
...
+
+      {{if has . "default_consent_ttl" -}}
+      default_consent_ttl = {
+        {{hcl .default_consent_ttl}}
+      }
+      {{end -}}
```
Comment (xingao267) [inline templates/tfengine/components/resources/healthcare_datasets/main.tf:43]: You can do {{hclField .default_consent_ttl}} for this.

File: docs/tfengine/schemas/resources.md [hunk 3]
```diff
@@ -752,6 +752,54 @@ Type: array(string)
...
+
+### healthcare_datasets.consent_stores.iam_members.role
+
+IAM role to grant.
+
+Type: string
+
+### healthcare_datasets.consent_stores.labels
+
+Labels to set on the consent store.
```
Comment (MartinPetkov) [inline docs/tfengine/schemas/resources.md:781]: Could you link some documentation on what these labels are? It seems like just a key-value pair but the documentation just says it's "object" and it's not obvious how it's structured.

File: examples/tfengine/generated/team/project_data/main.tf [hunk 4]
```diff
@@ -107,6 +107,17 @@ module "healthcare_dataset" {
   project  = module.project.project_id
   location = "us-central1"
 
+  consent_stores = [
+    {
+      name = "example-consent-store"
+      enable_consent_create_on_update = true
+      default_consent_ttl = "100000s"
```
Comment (MartinPetkov) [inline examples/tfengine/generated/team/project_data/main.tf:114]: Nit: Please add a comment with approximately how many days/hours 10000s is.

File: templates/tfengine/components/resources/healthcare_datasets/main.tf [hunk 5]
```diff
@@ -23,6 +23,37 @@ module "{{resourceName . "name"}}" {
```
Comment (xingao267) [inline templates/tfengine/components/resources/healthcare_datasets/main.tf:23]: Unrelated to this CL, but could you update the version from 1.3.0 to 2.0.0? I just made a release today

File: go.sum [hunk 6]
```diff
@@ -18,6 +18,7 @@ cloud.google.com/go v0.74.0 h1:kpgPA77kSSbjSs+fWHkPTxQ6J5Z2Qkruo5jfXEkHxNQ=
 cloud.google.com/go v0.74.0/go.mod h1:VV1xSbzvo+9QJOxLDaJfTjx5e+MePCpCWwvftOeQmWk=
 cloud.google.com/go v0.78.0 h1:oKpsiyKMfVpwR3zSAkQixGzlVE5ovitBuO0qSmCf0bI=
 cloud.google.com/go v0.78.0/go.mod h1:QjdrLG0uq+YwhjoVOLsS1t7TW8fs36kLs4XO5R5ECHg=
+cloud.google.com/go v0.79.0 h1:oqqswrt4x6b9OGBnNqdssxBl1xf0rSUNjU2BR4BZar0=
```
Comment (MartinPetkov) [inline go.sum:21]: The changes to the go.sum file seem unrelated to this PR. Please rebase or merge the `master` branch and if there are still changes to the go.sum file, open a separate PR just for them.

File: templates/tfengine/components/resources/healthcare_datasets/main.tf [hunk 7]
```diff
@@ -23,6 +23,37 @@ module "{{resourceName . "name"}}" {
...
+
+      {{if has . "default_consent_ttl" -}}
+      default_consent_ttl = {
+        {{hcl .default_consent_ttl}}
+      }
+      {{end -}}
+
+      {{if $labels := merge (get $ "labels") (get . "labels") -}}
+      labels = {
+        {{range $k, $v := $labels -}}
```
Comment (MartinPetkov) [inline templates/tfengine/components/resources/healthcare_datasets/main.tf:47]: Could you try using the `hcl` function here too? You should be able to do `{{hcl $labels}}`

**Comment by MartinPetkov** [comment]
From the failed presubmit job:
```
Please run the following command from the repo root and check in the changes:
./scripts/regen.sh
```

File: templates/tfengine/components/resources/healthcare_datasets/main.tf [hunk 8]
```diff
@@ -15,14 +15,43 @@ limitations under the License. */ -}}
...
+      name = "{{.name}}"
+      {{hclField . "iam_members" -}}
+
+      {{if has . "enable_consent_create_on_update" -}}
+      enable_consent_create_on_update = {
+        {{hclField .enable_consent_create_on_update}}
+      }
+      {{end -}}
+
+      {{if has . "default_consent_ttl" -}}
```
Comment (MartinPetkov) [inline templates/tfengine/components/resources/healthcare_datasets/main.tf:39]: Same here, replace with
```
{{hclField . "default_consent_ttl"}}
```

File: examples/tfengine/modules/team.hcl [hunk 9]
```diff
@@ -288,6 +288,12 @@ template "project_data" {
       }]
       healthcare_datasets = [{
         name = "healthcare-dataset"
+        consent_stores = [{
+          name = "consent-store"
+          labels = {
```
Comment (MartinPetkov) [inline examples/tfengine/modules/team.hcl:293]: Could you please add enable_consent_create_on_update and default_consent_ttl here so they show up in the example?

**Review by MartinPetkov — APPROVED** [review]

**Comment by MartinPetkov** [comment]
LGTM, but I'd like @xingao267 to look as well just in case.

**Review by xingao267 — APPROVED** [review]
